hadoop-yarn-project/hadoop-yarn/hadoop-yarn-server/hadoop-yarn-server-resourcemanager/src/main/java/org/apache/hadoop/yarn/server/resourcemanager/federation/FederationStateStoreService.java

```diff
@@ -259,378 +261,190 @@ public Version loadVersion() {
       DeleteReservationHomeSubClusterRequest request) throws YarnException {
-    try {
-      long startTime = clock.getTime();
-      DeleteReservationHomeSubClusterResponse response =
-          stateStoreClient.deleteReservationHomeSubCluster(request);
-      long stopTime = clock.getTime();
-      FederationStateStoreServiceMetrics.succeededStateStoreServiceCall(
-          stopTime - startTime);
-      return response;
-    } catch (YarnException e) {
-      LOG.error("deleteReservationHomeSubCluster error.", e);
-      FederationStateStoreServiceMetrics.failedStateStoreServiceCall();
-      throw e;
-    }
+    FederationClientMethod clientMethod = new FederationClientMethod(
+        "deleteReservationHomeSubCluster", DeleteReservationHomeSubClusterResponse.class, request);
+    return invoke(clientMethod, DeleteReservationHomeSubClusterResponse.class);
   }
 
   @Override
   public RouterMasterKeyResponse storeNewMasterKey(RouterMasterKeyRequest request)
       throws YarnException, IOException {
-    try {
-      long startTime = clock.getTime();
-      RouterMasterKeyResponse response = stateStoreClient.storeNewMasterKey(request);
-      long stopTime = clock.getTime();
-      FederationStateStoreServiceMetrics.succeededStateStoreServiceCall(
-          stopTime - startTime);
-      return response;
-    } catch (YarnException e) {
-      LOG.error("storeNewMasterKey error.", e);
-      FederationStateStoreServiceMetrics.failedStateStoreServiceCall();
-      throw e;
-    }
+    FederationClientMethod clientMethod = new FederationClientMethod(
+        "storeNewMasterKey", RouterMasterKeyResponse.class, request);
+    return invoke(clientMethod, RouterMasterKeyResponse.class);
   }
 
   @Override
   public RouterMasterKeyResponse removeStoredMasterKey(RouterMasterKeyRequest request)
       throws YarnException, IOException {
-    try {
-      long startTime = clock.getTime();
-      RouterMasterKeyResponse response = stateStoreClient.removeStoredMasterKey(request);
-      long stopTime = clock.getTime();
-      FederationStateStoreServiceMetrics.succeededStateStoreServiceCall(
-          stopTime - startTime);
-      return response;
-    } catch (YarnException e) {
-      LOG.error("removeStoredMasterKey error.", e);
-      FederationStateStoreServiceMetrics.failedStateStoreServiceCall();
-      throw e;
-    }
+    FederationClientMethod clientMethod = new FederationClientMethod(
+        "removeStoredMasterKey", RouterMasterKeyResponse.class, request);
+    return invoke(clientMethod, RouterMasterKeyResponse.class);
   }
 
   @Override
   public RouterMasterKeyResponse getMasterKeyByDelegationKey(RouterMasterKeyRequest request)
       throws YarnException, IOException {
+    FederationClientMethod clientMethod = new FederationClientMethod(
+        "getMasterKeyByDelegationKey", RouterMasterKeyResponse.class, request);
+    return invoke(clientMethod, RouterMasterKeyResponse.class);
```

Comment:
could we do something like:
```
clientMethod.invoke();
```

Reply:
Thank you for your suggestion, I will refactor this part of the code.